api/pricers/azure/vm.go

```diff
@@ -2,23 +2,40 @@ package azure
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"github.com/zparnold/azure-terraform-cost-estimator/common"
 	"k8s.io/klog/v2"
 	"strings"
 )
 
+type PriceType int
```

Comment:
Slight nit pick here, but how do you feel about calling this a `PricingScheme` and not a `PriceType`?

Reply:
I've renamed the enum to PricingScheme